release/mixer.sh

```diff
@@ -184,7 +184,7 @@ stage Mixer
 pushd "${BUILD_DIR}" > /dev/null
 
 section "Bootstrapping Mix Workspace"
-mixer init --upstream-url "${CLR_PUBLIC_DL_URL}"
+mixer init --upstream-url "${CLR_PUBLIC_DL_URL}" --upstream-version ${CLR_LATEST}
```

Comment:
Could you please elaborate on why this is needed? Because this is already done by 'generate_mix' and 'generate_bump' via 'mixer versions update'

Reply:
Sometimes, we need set the CLR_LATEST in config.sh to a certain one.  For example, in our case, I set CLR_LATEST as 29280. Also I used the internal FTP mirror to speed up which has some delay on the CLR updates. ( The latest in FTP mirror SH is 29310 but it is 29350 already)
if without this patch, I would meet below issue when doing mixer.
"
== Building ==

Regular Mix::
    29280010 (27) based on: 29280 (27)
generate_mix 111
2019/05/10 01:53:27 ERROR: couldn't download information about upstream: Failed to get format for version "29350": got status "404 Not Found" when downloading: http://linux-ftp.sh.intel.com/pub/mirrors/clearlinux/update/29350/format
make: *** [Makefile:35: mixer] Error 1
"
Passing the parameter upstream-version to mixer init could solve the issue above.

Reply:
You should not be setting CLR_LATEST via config.sh, unless it is really static information for you. CLR_LATEST is a mean to enforce a build will be done against a specific version of Clear, usually a temporary settings.

It can be set through config.sh though. But is usually a String Parameter in Jenkins that is left blank most of time.

Apart from that, we could set the upstream version on mixer init and again the same version on generate mix. But what is really broken in this case isn't that, it is your mirror. It should not update latest file before having the actual update in the server. That is the real issue.

Default operation of distro-factory is too have CLR_LATEST blank and letting the system figure out the latest available version. This would break anyway even with the proposed change.

Reply:
Yes, you are right about the mirror issue. 
The reason why I want one option to set the CLR_LATEST is, for some customers, they don't want to track all the Clear release version, don't always sync to the latest. They want to have their own release cadence, for example, once per week, sync to the Clear version they already has full validation on (probably not the latest).

Reply:
But that can be done. Permanently setting CLR_LATEST is a valid use case and there are people using bistro-factory this way. However, they are pointing to mirrors that correctly sync new releases (even that they keep their fork in a given version).

Reply:
But there is no harm to set --upstream-version as CLR_LATEST, right? And considering the mirror sync issue may not be fixed in a short time, this change can be helpful.